pkg/block/gs/dual_adapter.go

```diff
@@ -0,0 +1,166 @@
+package gs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/block"
+)
+
+const defaultLakeFSPrefix = "_lakefs/"
+
+var errCrossAdapterCopy = fmt.Errorf("cross adapter copy is not supported")
```

Comment:
make public and of type ErrNotSupported
```suggestion
var ErrCrossAdapterCopy = fmt.Errorf("cross adapter copy is not supported: %w", block.ErrOperationNotSupported)
```